### PR TITLE
mgr: make standby get_mgr_ip() use the right IP

### PR DESCRIPTION
**Pull request by liewegas**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mgr/BaseMgrStandbyModule.cc [hunk 1]
```diff
@@ -164,7 +165,48 @@ ceph_log(BaseMgrStandbyModule *self, PyObject *args)
...
+    entity_addrvec_t myaddrs = self->this_module->get_myaddrs();
+    with_gil_t with_gil{no_gil};
+    f.open_array_section("ips");
+    std::set<std::string> did;
+    for (auto& i : myaddrs.v) {
+      std::string ip = i.ip_only_to_str();
+      if (did.count(ip)) {
+	continue;
+      }
+      did.insert(ip);
```
Comment (tchaikov) [inline src/mgr/BaseMgrStandbyModule.cc:194]:
```suggestion
      if (auto [where, inserted] = did.insert(ip); inserted) {
	f.dump_string("ip", ip);
      }
```

File: src/mgr/BaseMgrStandbyModule.cc [hunk 2]
```diff
@@ -164,7 +165,48 @@ ceph_log(BaseMgrStandbyModule *self, PyObject *args)
...
 }
 
+static PyObject*
+ceph_standby_state_get(BaseMgrStandbyModule *self, PyObject *args)
+{
+  char *whatc = NULL;
+  if (!PyArg_ParseTuple(args, "s:ceph_state_get", &whatc)) {
+    return NULL;
+  }
+  std::string what(whatc);
```
Comment (tchaikov) [inline src/mgr/BaseMgrStandbyModule.cc:175]: nit, `std::string_view` is a better fit in this case.

**Review by tchaikov — APPROVED** [review]

**Review by Daniel-Pivonka — APPROVED** [review]
resolved errors seen https://tracker.ceph.com/issues/51446

**Review by tchaikov — APPROVED** [review]

**Review by p-se — APPROVED** [review]
Fixes my issue with running `run-backend-api-tests.sh`.

**Pull request merged**